client/src/templates/Challenges/redux/code-storage-epic.js

```diff
@@ -72,17 +74,44 @@ function clearCodeEpic(action$, state$) {
 }
 
 function saveCodeEpic(action$, state$) {
+  let saveSuccessful;
   return action$.pipe(
-    ofType(types.executeChallenge),
+    ofType(types.executeChallenge, types.saveEditorContent),
     // do not save challenge if code is locked
     filter(() => !isCodeLockedSelector(state$.value)),
     tap(() => {
       const state = state$.value;
       const { id } = challengeMetaSelector(state);
       const files = challengeFilesSelector(state);
-      store.set(id, files);
+      try {
+        store.set(id, files);
+        // Possible fileType values: indexhtml indexjs indexjsx
+        // The files Object always has one of these as the first/only attribute
+        const fileType = Object.keys(files)[0];
+        if (store.get(id)[fileType].contents !== files[fileType].contents) {
+          throw Error('Failed to save to localStorage');
+        }
+        saveSuccessful = true;
+      } catch (e) {
+        saveSuccessful = false;
+      }
     }),
-    ignoreElements()
+    ofType(types.saveEditorContent),
+    switchMap(() =>
+      saveSuccessful
```

Comment:
saveSuccess will always be set to true or false because the previous, 'tap()' function is executed first. If the localStorage is to full or if for whatever other reason the code did not save correctly, the value will be set to false and the user will be notified that their code was not saved successfully (if they pressed Ctrl + S).  I moved the logic for the actual flash creation into the switchMap() function because I don't think it makes sense to show the flash message when the user executes the challenge.   

 If their is a better way to do this please let me know! I do not work with rxjs much.